app/src/main/java/com/example/scoutconcordia/MapsActivity.java

```diff
@@ -282,6 +269,64 @@ else if (reason == GoogleMap.OnCameraMoveStartedListener.REASON_API_ANIMATION) {
         }
     }
 
+    // This is the method that will be used to encrypt all of the input files during the app startup.
+    public void encryptAllInputFiles()
+    {
+        InputStream fis = null;
+        OutputStream fos = null;
+        String filename = "";
+        String encrypted_filename = "";
+
+        try {
+            // encrypt hall 8th floor
+            filename = "hall8nodes";
+            encrypted_filename = "encrypted_hall8.txt";
+            fis = getResources().openRawResource(getResources().getIdentifier(filename, "raw", getPackageName()));
+            fos = new FileOutputStream(new File(MapsActivity.this.getFilesDir().getAbsoluteFile(), encrypted_filename));
+            encrypter.encryptFile(fis, fos);
+
+            // encrypt any other floor map here...
+
+
+            // close the input and the output streams
+            fis.close();
+            fos.close();
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    // this method will be used for creating the floor graphs by reading form a node encrypted text file.
+    public void createGraph(Graph graphName, String encryptedFileName)
```

Comment:
Having this method is great but it should be refactored to the graph class in the case that we need to call this method from another class / activity. Everything pertaining to the graph should be in the graph class / atomic